<?xml version="1.0"?>
<robot name="kuka">

    <link name="base_footprint"/>

    <joint name="fixed" type="fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
    </joint>

    <material name="Orange">
      <color rgba="1.0 0.4 0.0 1.0"/>
    </material>
    <material name="Grey">
      <color rgba="0.7 0.7 0.7 1.0"/>
    </material>
    <material name="DarkGrey">
      <color rgba="0.3 0.3 0.3 1.0"/>
    </material>
    <material name="White">
      <color rgba="1.0 1.0 1.0 1.0"/>
    </material>
    <material name="Black">
      <color rgba="0.0 0.0 0.0 1.0"/>
    </material>
    <material name="Red">
      <color rgba="1.0 0.0 0.0 1.0"/>
    </material>
    <material name="Green">
      <color rgba="0.0 1.0 0.0 1.0"/>
    </material>

    <link name="base_link">
        <visual>
          <geometry>
            <box size="1.08 0.63 0.085"/>
          </geometry>
          <origin rpy="0 0 0" xyz="0 0 -0.1885"/>
          <material name="Orange"/>
        </visual>
        <collision>
          <geometry>
            <box size="1.08 0.63 0.085"/>
          </geometry>
        </collision>
        <inertial>
           <mass value="364"/>
           <inertia ixx="25.86" ixy="0.0" ixz="0.0" iyy="47.42" iyz="0.0" izz="49.2"/>
         </inertial>
    </link>

    <joint name="base_bottom_to_middle_joint" type="fixed">
           <parent link="base_link" />
           <child link="base_link_middle" />
           <origin xyz="0 0 -0.122" rpy="0 0 0"/>
    </joint>

    <link name="base_link_middle">
        <visual>
          <geometry>
            <box size="0.84 0.48 0.048"/>
          </geometry>
          <origin rpy="0 0 0" xyz="0 0 0.0"/>
          <material name="Orange"/>
        </visual>
        <collision>
          <geometry>
            <box size="0.920 0.51 0.055"/>
          </geometry>
        </collision>
        <inertial>
           <mass value="364"/>
           <inertia ixx="25.86" ixy="0.0" ixz="0.0" iyy="47.42" iyz="0.0" izz="49.2"/>
        </inertial>
     </link>

    <joint name="base_middle_to_top_joint" type="fixed">
           <parent link="base_link_middle" />
           <child link="base_link_top" />
           <origin xyz="0 0 0.2865" rpy="0 0 0"/>
    </joint>

    <link name="base_link_top">
        <visual>
          <geometry>
            <box size="1.08 0.63 0.525"/>
          </geometry>
          <origin rpy="0 0 0" xyz="0 0 0.0"/>
          <material name="Grey"/>
        </visual>
        <collision>
          <geometry>
            <box size="1.08 0.63 0.5335"/>
          </geometry>
        </collision>
        <inertial>
           <mass value="364"/>
           <inertia ixx="25.86" ixy="0.0" ixz="0.0" iyy="47.42" iyz="0.0" izz="49.2"/>
         </inertial>
    </link>

    <joint name="laser_B1_joint" type="fixed">
           <parent link="base_link" />
           <child link="laser_B1_link" />
           <origin xyz="0.483 -0.258 -0.112" rpy="0 0 -0.7854"/>
    </joint>

    <link name="laser_B1_link">
       <visual>
         <origin xyz="0 0 0" rpy="3.1413 0 0"/>
         <geometry>
             <mesh filename="package://kuka_bringup/meshes/sensors/visual/sick_lms1xx.dae" />
         </geometry>
         <material name="dark"/>
       </visual>
       <collision>
           <origin xyz="0.0 0 0.0" rpy="0 0 0"/>
           <geometry>
               <mesh filename="package://kuka_bringup/meshes/sensors/collision/sick_lms1xx.stl" />
           </geometry>
       </collision>
    </link>

    <joint name="laser_B4_joint" type="fixed">
            <parent link="base_link" />
            <child link="laser_B4_link" />
            <origin xyz="-0.483 0.258 -0.112" rpy="0 0 2.3562"/>
    </joint>

    <link name="laser_B4_link">
        <visual>
          <origin xyz="0.0 0 0" rpy="0 3.1413 3.1413"/>
          <geometry>
              <mesh filename="package://kuka_bringup/meshes/sensors/visual/sick_lms1xx.dae" />
          </geometry>
          <material name="dark"/>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                  <mesh filename="package://kuka_bringup/meshes/sensors/collision/sick_lms1xx.stl" />
            </geometry>
        </collision>
    </link>

    <link name="front_left_wheel_link">
        <visual>
          <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/wheels/omni_wheel_1.dae" />
          </geometry>
        </visual>
        <collision>
          <geometry>
            <cylinder length="0.175" radius="0.125" />
          </geometry>
        </collision>

        <inertial>
          <mass value="6.5" />
          <origin xyz="0 0 0" />
          <inertia ixx="0.04197" ixy="0.0" ixz="0.0" iyy="0.0508" iyz="0.0" izz="0.04197"/>
        </inertial>
    </link>

    <link name="front_right_wheel_link">
        <visual>
          <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/wheels/omni_wheel_1.dae" />
          </geometry>
        </visual>
        <collision>
          <geometry>
            <cylinder length="0.175" radius="0.125" />
          </geometry>
        </collision>

        <inertial>
          <mass value="6.5" />
          <origin xyz="0 0 0" />
          <inertia ixx="0.04197" ixy="0.0" ixz="0.0" iyy="0.0508" iyz="0.0" izz="0.04197"/>
        </inertial>
    </link>

    <link name="back_left_wheel_link">
        <visual>
         <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/wheels/omni_wheel_1.dae" />
          </geometry>
        </visual>
        <collision>
          <geometry>
            <cylinder length="0.175" radius="0.125" />
          </geometry>
        </collision>

        <inertial>
          <mass value="6.5" />
          <origin xyz="0 0 0" />
          <inertia ixx="0.04197" ixy="0.0" ixz="0.0" iyy="0.0508" iyz="0.0" izz="0.04197"/>
        </inertial>
    </link>

    <link name="back_right_wheel_link">
        <visual>
          <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/wheels/omni_wheel_1.dae" />
          </geometry>
        </visual>
        <collision>
          <geometry>
            <cylinder length="0.175" radius="0.125" />
          </geometry>
        </collision>

        <inertial>
          <mass value="6.5" />
          <origin xyz="0 0 0" />
          <inertia ixx="0.04197" ixy="0.0" ixz="0.0" iyy="0.0508" iyz="0.0" izz="0.04197"/>
        </inertial>
    </link>

    <joint name="front_left_wheel_joint" type="fixed">
        <parent link="base_link"/>
        <child link="front_left_wheel_link"/>
        <axis xyz="0 1 0"/>
        <origin xyz="0.28 0.1825 -0.135"/>
        <limit effort="100" velocity="100"/>
        <joint_properties damping="0.0" friction="0.0"/>
    </joint>

    <joint name="front_right_wheel_joint" type="fixed">
        <parent link="base_link"/>
        <child link="front_right_wheel_link"/>
        <origin xyz="0.28 -0.1825 -0.135"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="100"/>
        <joint_properties damping="0.0" friction="0.0"/>
    </joint>

    <joint name="back_left_wheel_joint" type="fixed">
        <parent link="base_link"/>
        <child link="back_left_wheel_link"/>
        <origin xyz="-0.28 0.1825 -0.135"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="100"/>
        <joint_properties damping="0.0" friction="0.0"/>
    </joint>

    <joint name="back_right_wheel_joint" type="fixed">
        <parent link="base_link"/>
        <child link="back_right_wheel_link"/>
        <origin xyz="-0.28 -0.1825 -0.135"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="100"/>
        <joint_properties damping="0.0" friction="0.0"/>
    </joint>

        <joint name="base_link-to-iiwa" type="fixed">
        <origin rpy="0 0 -1.5708" xyz="0.363 -0.184 0.2625"/>
        <parent link="base_link_top"/>
        <child link="base_iiwa"/>
    </joint>

    <link name="base_iiwa">
        <visual>
          <origin rpy="0.0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/base_link.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/base_link.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_1">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_1.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_1.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_2">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_2.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_2.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_3">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_3.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_3.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_4">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_4.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_4.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_5">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_5.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_5.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_6">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_6.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_6.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="link_7">
        <visual>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/visual/link_7.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <mesh filename="package://kuka_bringup/meshes/lbr_iiwa_14_r820/collision/link_7.stl"/>
          </geometry>
        </collision>
    </link>
    <link name="tool0"/>

    <joint name="joint_a1" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="base_iiwa"/>
        <child link="link_1"/>
        <axis xyz="0 0 1"/>
        <limit effort="0" lower="-2.9668" upper="2.9668" velocity="1.4834"/>
    </joint>
    <joint name="joint_a2" type="fixed">
        <origin rpy="0 0 0" xyz="-0.00043624 0 0.36"/>
        <parent link="link_1"/>
        <child link="link_2"/>
        <axis xyz="0 1 0"/>
        <limit effort="0" lower="-2.0942" upper="2.0942" velocity="1.4834"/>
    </joint>
    <joint name="joint_a3" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="link_2"/>
        <child link="link_3"/>
        <axis xyz="0 0 1"/>
        <limit effort="0" lower="-2.9668" upper="2.9668" velocity="1.7452"/>
    </joint>
    <joint name="joint_a4" type="fixed">
        <origin rpy="0 0 0" xyz="0.00043624 0 0.42"/>
        <parent link="link_3"/>
        <child link="link_4"/>
        <axis xyz="0 -1 0"/>
        <limit effort="0" lower="-2.0942" upper="2.0942" velocity="1.3089"/>
    </joint>
    <joint name="joint_a5" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="link_4"/>
        <child link="link_5"/>
        <axis xyz="0 0 1"/>
        <limit effort="0" lower="-2.9668" upper="2.9668" velocity="2.2688"/>
    </joint>
    <joint name="joint_a6" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0.4"/>
        <parent link="link_5"/>
        <child link="link_6"/>
        <axis xyz="0 1 0"/>
        <limit effort="0" lower="-2.0942" upper="2.0942" velocity="2.356"/>
    </joint>
    <joint name="joint_a7" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="link_6"/>
        <child link="link_7"/>
        <axis xyz="0 0 1"/>
        <limit effort="0" lower="-3.0541" upper="3.0541" velocity="2.356"/>
    </joint>
    <joint name="joint_a7-tool0" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0.126"/>
        <parent link="link_7"/>
        <child link="tool0"/>
        <axis xyz="0 0 0"/>
    </joint>

</robot>
